src/core/Akka.Remote.Tests.Performance/InboundMessageDispatcherSpec.cs

```diff
@@ -0,0 +1,97 @@
+        private static readonly AtomicCounter Counter = new AtomicCounter(0);
+
+        private ActorSystem _actorSystem;
+        private Address _systemAddress;
+        private IInboundMessageDispatcher _dispatcher;
+
+        private IInternalActorRef _targetActorRef;
+
+        /// <summary>
+        /// Not thread-safe, but called by a single thread in the benchmark
+        /// </summary>
+        private class BenchmarkActorRef : MinimalActorRef
+        {
+            private readonly Counter _counter;
+
+            public BenchmarkActorRef(Counter counter)
+            {
+                _counter = counter;
+            }
+
+            protected override void TellInternal(object message, IActorRef sender)
+            {
+                _counter.Increment();
+            }
+
+            public override ActorPath Path { get { return null; } }
+            public override IActorRefProvider Provider { get { return null; } }
+        }
+
+        private static readonly Config RemoteHocon = ConfigurationFactory.ParseString(@"
+             akka {
+              actor.provider = ""Akka.Remote.RemoteActorRefProvider,Akka.Remote""
+
+              remote {
+                log-remote-lifecycle-events = off
+
+                enabled-transports = [
+                  ""akka.remote.test"",
+                ]
+
+                test {
+                  transport-class = ""Akka.Remote.Transport.TestTransport,Akka.Remote""
+                  applied-adapters = []
+                  registry-key = aX33k12WKg
+                  maximum-payload-bytes = 128000b
+                  scheme-identifier = test
+                  local-address = ""test://MessageDispatcher@0.0.0.0:1111""
+                }
+              }
+        ");
+
+        [PerfSetup]
+        public void Setup(BenchmarkContext context)
+        {
+            _actorSystem = ActorSystem.Create("MessageDispatcher" + Counter.GetAndIncrement(), RemoteHocon);
+            _systemAddress = RARP.For(_actorSystem).Provider.DefaultAddress;
+            _inboundMessageDispatcherCounter = context.GetCounter(MessageDispatcherThroughputCounterName);
+            _message = SerializedMessage.CreateBuilder().SetSerializerId(0).SetMessage(ByteString.CopyFromUtf8("foo")).Build();
+            _dispatcher = new DefaultMessageDispatcher(_actorSystem, RARP.For(_actorSystem).Provider, _actorSystem.Log);
+            _targetActorRef = new BenchmarkActorRef(_inboundMessageDispatcherCounter);
+        }
+
+        [PerfBenchmark(Description = "Tests the performance of the Default", RunMode = RunMode.Throughput, NumberOfIterations = 13, TestMode = TestMode.Measurement)]
+        [CounterMeasurement(MessageDispatcherThroughputCounterName)]
+        public void DispatchThroughput(BenchmarkContext context)
+        {
+            _dispatcher.Dispatch(_targetActorRef, _systemAddress, _message);
+        }
+
+        [PerfCleanup]
+        public void Cleanup()
+        {
+            _actorSystem.Shutdown();
```

Comment:
Needed an `ActorSystem` in order to satisfy the constructor requirements of the `DefaultMessageDispatcher`, and it also had to have Akka.Remote enabled with a transport since the `DefaultMessageDispatcher` depends on being able to get a reference to a `RemoteActorRefProvider` internally.